src/machines/java/com/enderio/machines/common/blockentity/base/PoweredMachineEntity.java

```diff
@@ -94,12 +96,25 @@ public void serverTick() {
      * On client side, this will likely be an instance of {@link ImmutableMachineEnergyStorage}.
      * On server side, it will be an instance descended of {@link MachineEnergyStorage}.
      */
-    public IMachineEnergyStorage getEnergyStorage() {
+    public final IMachineEnergyStorage getEnergyStorage() {
         if (isClientSide()) {
             return clientEnergyStorage;
         }
         return energyStorage;
     }
+    /**
+     * Get the machine's exposed energy storage. Will likely be the same as the normal energy storage, but sometimes might be different, if blocks want to expose a different energystorage (wrapper for combining photovoltaic cells or capacitor banks)
+     */
+    public final IMachineEnergyStorage getExposedEnergyStorage() {
```

Comment:
Sorry, my suggestion wasn't fully clear.
Instead of having another field (in this case exposedEnergyStorage), we can make this non-final and by default just forward onto getEnergyStorage() but then can be overridden by solar panels :)

Reply:
in short: I understood what you wanted, but it was not doable in a good way, so this was the best/closest I could do

Reply:
No worries at all! I'm happy with this then :) Nice work!